### ISSUE TITLE
What's the best way to extend the entrypoint?

### ISSUE THREAD

**Issue by timwis** [docker-library/wordpress#316]
I need run `npm install`, `npm run build`, and add a few settings to `wp-config.php`. I also need to fetch private plugins, but that part I could probably get away with doing in the `Dockerfile`. And of course during development I want to run `npm install` and `npm run build:watch`, so I'll need to leverage an entrypoint.

What's the best way to extend the existing entrypoint? I've made my own `entrypoint.sh` file (different filename so it doesn't conflict) and am calling the original one at the end via `exec /usr/local/bin/docker-entrypoint.sh "apache2-foreground"`. This works fine if I vendor wordpress in my repo, but if my repo only contains the source, then the wordpress docker image's entrypoint installs wordpress at _runtime_, and overwrites my `wp-config.php` changes. Alternatively I could move the `exec ....docker-entrypoint.sh` line to the top of my entrypoint, but then it hangs since apache runs in the foreground (which I imagine I want so I can stop it with ctrl+c when using compose).

Are there any examples of people using this image with their own theme and configurations that might make this more clear?

**Comment by tianon** [comment]
If `wp-config.php` exists already, the existing entrypoint shouldn't touch it:

https://github.com/docker-library/wordpress/blob/b7198b18d92c016411c4bc3cdb31711065305605/docker-entrypoint.sh#L130

Alternatively, you could make your changes directly to `wp-config-sample.php`, which is what our script copies.  I think your current approach is probably the best you're going to get.

**Comment by tianon** [comment]
It'll do even less so if you don't provide any of the environment variables it uses:

https://github.com/docker-library/wordpress/blob/b7198b18d92c016411c4bc3cdb31711065305605/docker-entrypoint.sh#L118-L119

**Comment by timwis** [comment]
I started copying over wordpress from `/usr/src/wordpress` to `/var/www/html` in my dockerfile so my entrypoint could run wp cli config commands and such, but the `VOLUME` in the worpress container made it so all my changes to that directory were discarded. I ended up just extending php-fpm directly :/

May I suggest that the wordpress container put wordpress into `/var/www/html` _in the image_ rather than at the entrypoint? The latter seems a bit magical and, it turns out, is really hard to work with. Forgive me if a lot of thought went into the reason behind this that I'm not aware of.
